feat: update random number generation function in `triangular/logpdf`

0PrashantYadav0 opened the pull request:
Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-    update random number generation function in `triangular/logpdf`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C implementation for triangular logpdf

0PrashantYadav0 committed:
chore: made changes as per code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

0PrashantYadav0 committed:
chore: merge develop

0PrashantYadav0 committed:
fix: changes in benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

0PrashantYadav0 committed:
chore: updated random generation function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/README.md:
```diff
@@ -145,10 +145,10 @@ var y;
-    c = a + ((b-a) * randu());
+    x = uniform( 0.0, 30.0 );
+    a = uniform( 0.0, 10.0 );
+    b = a + ( uniform( 0.0, 40.0 ) );
```
Here and elsewhere in this PR, we want to avoid adding several numbers together and prefer to just draw from the appropriate distribution. This can be rewritten more succinctly as

```suggestion
    b = uniform( a, a + 40.0 );
```

Planeshifter reviewed (changes requested):
Using `@stdlib/random/base/uniform` allows us to simplify the code by directly drawing from the desired distribution. Would be good to update accordingly.

0PrashantYadav0 commented:
Sure i think we need to implement `@stdlib/random/base/uniform` in all the distributions.

0PrashantYadav0 committed:
chore: update according to code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

0PrashantYadav0 commented:
@Planeshifter I made the changes please review and if all ok i will open another pr introducing the `@stdlib/random/base/uniform` for all the distributions.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/benchmark/benchmark.js:
```diff
@@ -84,7 +83,7 @@ bench( pkg+':factory', function benchmark( b ) {
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = ( randu()*2.0 ) - 2.0;
+		x = uniform( -2.0, 2.0 );
```
Can you allocate an array of values outside of the loop and then do `mylogpdf( [ x%x.length ] )`, rather than generate individual values within the loop?

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/benchmark/benchmark.js:
```diff
@@ -45,12 +45,11 @@ bench( pkg, function benchmark( b ) {
-		mode[ i ] = min[ i ] + ( ( max[ i ] - min[ i ] ) * randu() );
+		x[ i ] = uniform( 0.0, 30.0 );
+		min[ i ] = uniform( 0.0, 10.0 );
+		max[ i ] = uniform( min[ i ] + EPS, 40.0+min[ i ] + EPS );
```
```suggestion
		max[ i ] = uniform( min[ i ] + EPS, 40.0+ EPS + min[ i ] );
```

Be consistent with `native.js`.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/examples/index.js:
```diff
@@ -29,10 +29,10 @@ var y;
-	c = a + ((b-a) * randu());
+	x = uniform( 0.0, 30.0 );
+	a = uniform( 0.0, 10.0 );
+	b = a + ( uniform( 0.0, 40.0 ) );
```
This doesn't match the README.

0PrashantYadav0 committed:
fix: improve random number generation in triangular logpdf benchmark and examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte @Planeshifter Made all the changes you asked for please review and merge.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/benchmark/benchmark.js:
```diff
@@ -81,11 +81,14 @@ bench( pkg+':factory', function benchmark( b ) {
 	for ( i = 0; i < b.iterations; i++ ) {
-		x = ( randu()*2.0 ) - 2.0;
-		y = mylogpdf( x );
+		y = mylogpdf( x[ x%x.length ] );
```
```suggestion
		y = mylogpdf( x[ i%x.length ] );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/triangular/logpdf/benchmark/benchmark.js:
```diff
@@ -81,11 +81,14 @@ bench( pkg+':factory', function benchmark( b ) {
 	max = 1.5;
 	mode = 0.5;
 	mylogpdf = logpdf.factory( min, max, mode );
+	x = new Float64Array( len );
```
Where is `len` getting assigned a value?

Planeshifter reviewed (changes requested):
`benchmark.js` file might need some changes.

0PrashantYadav0 committed:
chore: changes in benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

0PrashantYadav0 commented:
@Planeshifter  I think i made the required changes if i miss something please do let me know. Thank you.

Planeshifter reviewed (approved):
LGTM.
